crates/emitter/src/scope_notes.rs

```diff
@@ -62,9 +62,36 @@ impl ScopeNoteList {
         ScopeNoteIndex::new(note_index)
     }
 
+    fn get_scope_hole_gcthing_index(&self, reference_scope_index: ScopeNoteIndex) -> GCThingIndex {
+        if reference_scope_index.index >= 1 {
+            return self
+                .notes
+                .get(reference_scope_index.index - 1)
+                .unwrap()
+                .index;
+        }
+        GCThingIndex::new(0)
+    }
+
+    pub fn enter_scope_hole(
+        &mut self,
+        reference_scope_index: ScopeNoteIndex,
+        offset: BytecodeOffset,
+    ) -> ScopeNoteIndex {
+        let note_index = self.notes.len();
+        let parent_index = ScopeNoteIndex::new(note_index - 1);
+        let reference_gcthing_index = self.get_scope_hole_gcthing_index(reference_scope_index);
+        self.enter_scope(reference_gcthing_index, offset, Some(parent_index))
+    }
+
     pub fn leave_scope(&mut self, index: ScopeNoteIndex, offset: BytecodeOffset) {
         self.notes[usize::from(index)].end = offset;
     }
+
+    pub fn current_index(&self) -> ScopeNoteIndex {
```

Comment:
this is returning something non-existent at this point and has no guarantee what it becomes,
IMO this method shouldn't exist
(or, if this is really necessary, this needs special comment and special name)